### PR TITLE
Feature/3900 error message

### PR DESCRIPTION
**Pull request by joannasia**
## Description

- Error message on form submit with fix button.

**New stuff** ✨

* Message box and handle fix function.

**Changes** 🏗

* Changes user with search button to one element for better accessibility.

(Resolves | Contributes to) #3900

### CODE REVIEW

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Hey @joannasia, nice work!

I spotted this issue:

https://user-images.githubusercontent.com/112586815/195214106-afd780f0-3db2-4ab7-ae49-756bc118982b.mov

Because the first invalid element (slider) isn't an input, it fires the custom event to which the user picker listens to. Initially I thought you'd need to add a check to the listener to see if the element is invalid but then I this scenario came to my mind:

https://user-images.githubusercontent.com/112586815/195214986-4ed04686-f82f-45f0-8dec-437434739040.mov

You might have to include some kind of an identifier in the event payload.

As a side note, I think it'd be nice to extract the UserPickerWithSearch's useEffect into a separate hook for better reusability with other fields that aren't inputs but I'll leave that decision to you.

File: src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx:83]: i asked about it @agnieszkajarosikloj, because this is the part of the code that was only moved by me, and she informed me that on other branch it is already changed and fixed

File: src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css
Comment (joannasia) [reply src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css]: span is not needed, it was during the process, it is removed now

File: src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.tsx
Comment (joannasia) [reply src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.tsx:216]: true, disable is enough, fixed

File: src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.css
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.css:5]: fits much better, it is changed, thanks

File: src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/Stages/FormStages.tsx:67]: yes, it does, it is fixed

**Comment by joannasia** [comment]
> 

i have added details info in this custom event, to make sure the proper element is taken and moved the useEffect to separate file

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Thanks for making the changes @joannasia! 
I think all looks good right now, I'm just wondering if we can improve the overall UX a bit. Right now, the number of errors doesn't update as you go and resolve them. It only refreshes when you click on Submit draft or Fix it. Would it be possible to update it more dynamically?

File: src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css
Comment (joannasia) [reply src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css]: it is changed

File: src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css
Comment (joannasia) [reply src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.css]: yes. thanks! it is much better

**Comment by joannasia** [comment]
@jakubcolony I have added the observer for the fields, to update the error messages dynamically

**Review by jakubcolony — APPROVED** [review]
Works really nice now @joannasia! Great job!
Please make sure you received at least 2 approvals before merging your PR

File: src/modules/dashboard/components/ExpenditurePage/Stages/constants.ts
Comment (joannasia) [reply src/modules/dashboard/components/ExpenditurePage/Stages/constants.ts]: moved it, thanks! Do you know maybe who I should trigger for CR to get the second approval ?

File: src/modules/pages/components/ExpenditurePage/constants.ts
Comment (chinins) [inline src/modules/pages/components/ExpenditurePage/constants.ts:14]: This is better to have in snake case `FIX_TRIGGER_EVENT_NAME`, that's our default fro constants like this.

**Review by chinins — APPROVED** [review]
Very good, just update the constant name please as per my separate comment.

File: src/modules/dashboard/components/ExpenditurePage/Stages/constants.ts
Comment (jakubcolony) [reply src/modules/dashboard/components/ExpenditurePage/Stages/constants.ts]: As long as you requested a review from `frontend` (which I see you did), others will see there's a pending review

**Pull request merged**
